pkg/platform/provider/baremetal/phases/image/image.go

```diff
@@ -0,0 +1,58 @@
+/*
+ * Tencent is pleased to support the open source community by making TKEStack
+ * available.
+ *
+ * Copyright (C) 2012-2019 Tencent. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * https://opensource.org/licenses/Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package image
+
+import (
+	"fmt"
+	"strings"
+
+	"tkestack.io/tke/pkg/util/log"
+
+	"tkestack.io/tke/pkg/platform/provider/baremetal/images"
+	"tkestack.io/tke/pkg/util/ssh"
+)
+
+type Option struct {
+	Version string
+}
+
+func Install(s ssh.Interface, option *Option) error {
+	images := images.ListCoreImageFullNamesWithK8sVerion(option.Version)
+	if len(images) == 0 {
+		return fmt.Errorf("images is empty")
+	}
+
+	log.Debug(fmt.Sprintf("core images:%v", images))
+	for _, image := range images {
+		cmd := fmt.Sprintf("docker pull %s", image)
+		_, err := s.CombinedOutput(cmd)
+		if err != nil {
+			if strings.Contains(err.Error(), "502 Bad Gateway") {
+				cmd = fmt.Sprintf(" docker info | grep Proxy")
+				output, _ := s.CombinedOutput(cmd)
+				return fmt.Errorf("pull image fail: %s. maybe docker proxy config err. docker info:%s. "+
```

Comment:
optimize error message

maybe set no_proxy for registry(%s) in docker dameon

domain has two form, use * for match will be better.

Reply:
https://docs.docker.com/config/daemon/systemd/#httphttps-proxy